test/Window/VideoMode.cpp

```diff
@@ -0,0 +1,100 @@
+#include <SFML/Window/VideoMode.hpp>
+
+#include <doctest.h>
+#include <ostream>
+
+namespace sf
+{
+std::ostream& operator <<(std::ostream& os, const sf::VideoMode& videoMode)
+{
+    os << videoMode.width << " x " << videoMode.height << " x " << videoMode.bitsPerPixel;
+    return os;
+}
+}
+
+TEST_CASE("sf::VideoMode class - [window]")
+{
+    SUBCASE("Construction")
+    {
+        SUBCASE("Default constructor")
+        {
+            sf::VideoMode videoMode;
+            CHECK(videoMode.width == 0);
+            CHECK(videoMode.height == 0);
+            CHECK(videoMode.bitsPerPixel == 0);
+        }
+
+        SUBCASE("Width, height constructor")
+        {
+            sf::VideoMode videoMode(800, 600);
+            CHECK(videoMode.width == 800);
+            CHECK(videoMode.height == 600);
+            CHECK(videoMode.bitsPerPixel == 32);
+        }
+
+        SUBCASE("Width, height, bit depth constructor")
+        {
+            sf::VideoMode videoMode(800, 600, 24);
+            CHECK(videoMode.width == 800);
+            CHECK(videoMode.height == 600);
+            CHECK(videoMode.bitsPerPixel == 24);
+        }
+    }
+
+    SUBCASE("Operators")
+    {
+        SUBCASE("operator==")
+        {
+            CHECK(sf::VideoMode() == sf::VideoMode());
+            CHECK(sf::VideoMode(0, 0, 0) == sf::VideoMode(0, 0, 0));
+            CHECK(sf::VideoMode(1080, 1920, 64) == sf::VideoMode(1080, 1920, 64));
+        }
+
+        SUBCASE("operator!=")
+        {
+            CHECK(sf::VideoMode() != sf::VideoMode(1, 0));
+            CHECK(sf::VideoMode() != sf::VideoMode(0, 1));
+            CHECK(sf::VideoMode() != sf::VideoMode(0, 0, 1));
+            CHECK(sf::VideoMode(720, 720) != sf::VideoMode(720, 720, 24));
+            CHECK(sf::VideoMode(1080, 1920, 16) != sf::VideoMode(400, 600));
+        }
+
+        SUBCASE("operator<")
+        {
+            CHECK(sf::VideoMode() < sf::VideoMode(0, 0, 1));
+            CHECK(sf::VideoMode(1080, 1920, 24) < sf::VideoMode(800, 800, 48));
+            CHECK(sf::VideoMode(400, 400, 48) < sf::VideoMode(400, 600, 48));
+            CHECK(sf::VideoMode(400, 400, 48) < sf::VideoMode(600, 400, 48));
```

Comment:
[Our `operator<`](https://github.com/SFML/SFML/blob/master/src/SFML/Window/VideoMode.cpp#L106-L123) considers, in this order:
* bits per pixel
* width
* height

So we should have the following test cases, which all resolve to `true` for `operator<`:
* bits less, width greater, height greater
* bits equal, width less, height greater
* bits equal, width equal, height less

And maybe we could additionally check that two equal ones are never less-than each other. Or there might even be a predefined way to test ordering relations for their consistency (e.g. what `std::map` does when using them as keys).

Same applies for other operators.

Reply:
This is great feedback. I tweaked the `operator<` and `operator>` tests to include these operations. I included these improved assertions in the `operator<=` and `operator>=` tests as well.